tests/test_utils/codec.cpp

```diff
@@ -0,0 +1,103 @@
+#include "test_utils/codec.hpp"
+
+#include <vector>
+
+#include <cbv2g/app_handshake/appHand_Decoder.h>
+#include <cbv2g/app_handshake/appHand_Encoder.h>
+
+#include <cbv2g/din/din_msgDefDecoder.h>
+#include <cbv2g/din/din_msgDefEncoder.h>
+
+#include <cbv2g/iso_20/iso20_AC_Decoder.h>
+#include <cbv2g/iso_20/iso20_AC_Encoder.h>
+#include <cbv2g/iso_20/iso20_DC_Decoder.h>
+#include <cbv2g/iso_20/iso20_DC_Encoder.h>
+
+namespace test_utils {
+
+template <typename DocType>
+static EncodingResult encode(int (*encode_func)(exi_bitstream_t*, DocType*), const DocType& request,
+                             const uint8_t* compare_data, std::size_t length) {
+    // FIXME (aw): what general size to take here?
+    uint8_t stream[256] = {};
+    exi_bitstream_t exi_stream_in;
+    size_t pos1 = 0;
+
+    exi_bitstream_init(&exi_stream_in, stream, sizeof(stream), pos1, nullptr);
+
+    if (0 != encode_func(&exi_stream_in, const_cast<DocType*>(&request))) {
+        return {false, false};
+    }
+
+    // FIXME (aw): why +1?
+    const auto encoded_stream = std::vector<uint8_t>(stream, stream + exi_bitstream_get_length(&exi_stream_in) + 1);
+
+    const auto expected_exi_stream = std::vector<uint8_t>(compare_data, compare_data + length);
```

Comment:
Good point :+1: , then I would propose we should either use
- `strlen` instead of `sizeof` or
- don't use a string literal, but rather initializing with a real array like `uint8_t doc_raw[] = {0x00, 0xf3, 0xab .. }`

Concerning the use of auto here, I don't have a strong opinion on that but I know that some people prefer this style because it is easier to infer the variable name (so first name, then type).

Reply:
I just looked at it too. Moritz, great that you found that out :+1: 
Then I would be in favor of a real array.

Reply:
Done

Reply:
I have no strong opinion about the `auto` style. Feel free to leave it as it is.